sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -130,7 +130,11 @@ case class AdaptiveSparkPlanExec(
     }
   }
 
-  @transient private val costEvaluator = SimpleCostEvaluator
+  @transient private val costEvaluator =
+    conf.getConf(SQLConf.ADAPTIVE_CUSTOM_COST_EVALUATOR_CLASS) match {
```

Comment:
Hey, how do we use this? `CostEvaluator` isn't an API.

Reply:
@HyukjinKwon - good callout. If the whole execution package is for internal purpose (I think the only public one is around data source, right?), how about moving `Cost` and `CostEvaluator` to `org.apache.spark.sql.catalyst.planning` or `org.apache.spark.sql.catalyst.plans.physical`?

Reply:
I'm okay with that.

Reply:
but the cost stuff should live under physical plans .. I would just mark it as `@Unstable` for now and unblock this PR.

Reply:
hey sorry I left some comments and removed it back. Yes, it makes sense to me.

Reply:
`catalyst` is also an internal package. 

`SparkSessionExtensions` exposes `LogicalPlan` as well, and I think unstable developer API does not have a hard requirement on the package names. I'm fine to add `@Unstable`

Reply:
Yeah, we can add `@Unstable` for now but I would also add a note that this class is supposed to be moved or changed in the near future.

Reply:
Sounds good, updated with `@Unstable` and added a note comment for class `Cost` and `CostEvaluator`.